misk-service/src/test/kotlin/misk/ServiceManagerModuleTest.kt

```diff
@@ -140,4 +142,71 @@ internal class ServiceManagerModuleTest {
         "misk.ServiceManagerModuleTest\$NonSingletonService2"
     )
   }
+
+  @Singleton
+  class ProducerService @Inject constructor(
+    private val log: StringBuilder
+  ) : AbstractService() {
+    init {
+      log.append("ProducerService.init\n")
+    }
+    override fun doStart() {
+      log.append("ProducerService.startUp\n")
+      notifyStarted()
+    }
+
+    override fun doStop() {
+      log.append("ProducerService.shutDown\n")
+      notifyStopped()
+    }
+  }
+
+  @Singleton
+  class ConsumerService @Inject constructor(
+    private val log: StringBuilder
+  ) : AbstractService() {
+    init {
+      log.append("ConsumerService.init\n")
+    }
+    override fun doStart() {
+      log.append("ConsumerService.startUp\n")
+      notifyStarted()
+    }
+
+    override fun doStop() {
```

Comment:
Just changed this to use AbstractService instead of AbstractIdleService because AbstractIdleService is concurrent and was flaking in CI.